manta/src/main/java/ch/cyberduck/core/manta/MantaSession.java

```diff
@@ -0,0 +1,192 @@
+import ch.cyberduck.core.LoginCallback;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.UrlProvider;
+import ch.cyberduck.core.exception.AccessDeniedException;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.exception.LoginFailureException;
+import ch.cyberduck.core.features.AttributesFinder;
+import ch.cyberduck.core.features.Delete;
+import ch.cyberduck.core.features.Directory;
+import ch.cyberduck.core.features.Home;
+import ch.cyberduck.core.features.Move;
+import ch.cyberduck.core.features.MultipartWrite;
+import ch.cyberduck.core.features.Read;
+import ch.cyberduck.core.features.Touch;
+import ch.cyberduck.core.features.Write;
+import ch.cyberduck.core.ssl.DefaultX509KeyManager;
+import ch.cyberduck.core.ssl.DisabledX509TrustManager;
+import ch.cyberduck.core.ssl.SSLSession;
+import ch.cyberduck.core.ssl.ThreadLocalHostnameDelegatingTrustManager;
+import ch.cyberduck.core.ssl.X509KeyManager;
+import ch.cyberduck.core.ssl.X509TrustManager;
+import ch.cyberduck.core.threading.CancelCallback;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+
+import com.joyent.manta.client.MantaClient;
+import com.joyent.manta.config.BaseChainedConfigContext;
+import com.joyent.manta.config.ChainedConfigContext;
+import com.joyent.manta.config.DefaultsConfigContext;
+import com.joyent.manta.config.SettableConfigContext;
+import com.joyent.manta.config.StandardConfigContext;
+
+public class MantaSession extends SSLSession<MantaClient> {
+
+    public static final Logger log = Logger.getLogger(MantaSession.class);
+
+    static final String HEADER_KEY_STORAGE_CLASS = "Durability-Level";
+
+    private final SettableConfigContext<BaseChainedConfigContext> config;
+
+    final MantaExceptionMappingService exceptionMapper;
+
+    private String keyFingerprint;
+
+    MantaPathMapper pathMapper;
+
+    public MantaSession(final Host h) {
+        this(h, new DisabledX509TrustManager(), new DefaultX509KeyManager());
+    }
+
+    public MantaSession(final Host h, final X509TrustManager trust, final X509KeyManager key) {
+        super(h, new ThreadLocalHostnameDelegatingTrustManager(trust, h.getHostname()), key);
+
+        pathMapper = new MantaPathMapper(this);
+        exceptionMapper = new MantaExceptionMappingService(this);
+        config = new ChainedConfigContext(
+                new DefaultsConfigContext(),
+                new StandardConfigContext()
+                        .setHttpsProtocols(DefaultsConfigContext.DEFAULT_HTTPS_PROTOCOLS)
+                        .setDisableNativeSignatures(true)
+                        .setNoAuth(false)
+                        .setMantaURL("https://" + h.getHostname())
+                        .setMantaUser(host.getCredentials().getUsername()));
+    }
+
+    @Override
+    protected MantaClient connect(final HostKeyCallback key) throws BackgroundException {
+        return null;
```

Comment:
The `MantaClient` constructor expects to be passed a usable `ConfigContext` and will throw an exception for the missing parameter values:

```
Errors when loading Manta SDK configuration:
Manta key id must be specified
Either the Manta key path must be set or the private key content must be set. Both can't be null.
```

I've been meaning to propose a restructuring of the way configs are prepared since it's not obvious that configuration values passed to the `MantaClient` instance cannot be changed after initialization. May I ask about the purpose of returning the client from `connect(HostKeyCallback)`?